### DISCUSSION TITLE
1 esp 3 relay 3 Fan speed

### DISCUSSION THREAD

**Discussion by nusf** [RavenSystem/esp-homekit-devices#1713] [MEPLHAA Scripts]
Hi, I'm really noob in coding. Trying to make JSON for kitchen fun with 3 speed steps, but stuck....

 {
"c":{"l":13,"b":[{"g":0,"t":5}]},
"a":
[{
"t":65,
"e":3,
"y0":[
{"v":1,"0":{"r":[{"g":12,"v":1}]}},
{"v":33,"0":{"r":[{"g":13,"v":1}]}},
{"v":67,"0":{"r":[{"g":14,"v":1}]}}]
}]
}

This JSON not working.

I made like 3 interlock switches

{
"c":{"l":13,"b":[{"g":0,"t":5}]},
"a":[{
"0":{"r":[{"g":12}]},
"1":{"r":[{"g":12,"v":1}],
"m":[[2],[3]]}
},{
"0":{"r":[{"g":13}]},
"1":{"r":[{"g":13,"v":1}],
"m":[[1],[3]]}
},{
"0":{"r":[{"g":14}]},
"1":{"r":[{"g":14,"v":1}],
"m":[[1],[2]]}
}]
}

, but with "t":65 and "e":3 it would be more beautiful.

Can some one help me.
Thank you
![IMG_1674](https://user-images.githubusercontent.com/61291086/151820923-baa8bc6b-1afd-4c97-a77d-04d7d29677a2.jpg)
.

**Comment by joddye2** [comment]
Check it out with translator:
https://eljokosan.wordpress.com/2021/07/04/homekit-haa-ventillator/

JSON with wildcard actions will work

**Reply by nusf** [reply]
Thx, this work like a charm!

**Reply by joddye2** [reply]
Good to hear. Your welcome!

**Reply by nusf** [reply]
Hello again! All this days I was trying to add buttons to activate 3 lvl speed 1 light and 1 button for everything off
I can't realise how to do this. Using NodeMCU or Memos D1 mini

This is my json:

{
  "c":{"l":13,"b":[{"g":0,"t":5}]},
  "a":[
    {
    "t":65,
    "e":3,
    "0":{"r":[{"g":5,"v":1},{"g":12,"v":1},{"g":14,"v":1}]},
    "y0":[
      {"v":1, "0":{"r":[{"g":5,"v":0}, {"g":12,"v":1}, {"g":14,"v":1}]},"b":[{"g":3}] },
      {"v":2, "0":{"r":[{"g":5,"v":1}, {"g":12,"v":1}, {"g":14,"v":0}]},"b":[{"g":16}] },
      {"v":3, "0":{"r":[{"g":5,"v":1}, {"g":12,"v":0}, {"g":14,"v":1}]},"b":[{"g":2}] }
    ]
  },{
    "0":{"r":[{"g":4,"v":1}]},
    "1":{"r":[{"g":4}]},
    "b":[{"g":15}]
  }
  
  ]
}

Button og gpio 15 working, but gpio 3, 16, 2 not working
And where to add button to off everything?

Please help.

**Comment by joddye2** [comment]
Hello!

Can you post a wiring of your project?

Do you want only one button to turn off it (and a led for off state)?

State imputs, according to the wiki:
- "f0" | Perform action "0" (Off)
- something like this:
      "f0": [ {"g": 15, "t": 1}]

I didn't attached buttons for states, but if I can, I'll help

**Reply by nusf** [reply]
I think I'll do this without led's
Just need buttons for my wife ) to activate 1, 2, 3 fan speed; light; and 1 button to off everything.

<a href="https://ibb.co/vzVptpP"><img src="https://i.ibb.co/vzVptpP/2022-02-19-00-07-09.png" alt="2022-02-19-00-07-09" border="0"></a>

<a href="https://ibb.co/dcJ5cDN"><img src="https://i.ibb.co/dcJ5cDN/IMG-1674.jpg" alt="IMG-1674" border="0"></a>

**Comment by joddye2** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Fan
- state imputs
- - f0 for off, f1 for on state supported

I think states for 1-2-3 not.

Probably solution.
Make "Switch" typed stateless buttons with the button inputs and then make automations.
e.g.: If switch1 turns on Then set fan speed to 33%.
